### ISSUE TITLE
Unfinished inline citation (?? something) hangs the script

### ISSUE THREAD

**Issue by hazzik** [FokkeZB/J2M#31]

**Comment by FokkeZB** [comment]
Could you provide an example?

**Comment by hazzik** [comment]
Just go to the website and type “?? Anything”

**Comment by FokkeZB** [comment]
Sorry, can't reproduce hanging the script:

![](https://cl.ly/182i323o1k0e/Screen%20Recording%202017-10-17%20at%2008.11%20AM.gif)

**Comment by hazzik** [comment]
I'm able to reproduce this constantly. Thee are several conditions:

# `??` shall be at the beginning of a line
# It shall be not at the end of the file
![ezgif-3-255a5087d6 1](https://user-images.githubusercontent.com/144791/31651610-e5db2a86-b378-11e7-8457-769d1883bfd5.gif)

**Comment by hazzik** [comment]
Easiest way to reproduce is
1. Open http://j2m.fokkezb.nl/
2. On line 15 select "citation??" (without first ??")
3. Press "delete"/"backspace"

**Comment by hazzik** [comment]
I can constantly reproduce this in Chrome/Firefox on Windows 10 and MacOS and on node 8.7
It does not hang in Safari and Mobile Safari

**Comment by FokkeZB** [comment]
Yep, with that I can reproduce, so it seems the issue is related to other stuff in the editor as well.

Unfortunately it indeed hangs, with no logs to aid debugging it further.

I'm also kind of iceboxing this project. If you have time to dig in and send a PR that would be appreciated.

**Comment by hazzik** [comment]
Here is a js file to reproduce:

[test.zip](https://github.com/FokkeZB/J2M/files/1390167/test.zip)

**Comment by hazzik** [comment]
The culprit is at regex which parses the citation.
